tests/components/smarla/test_config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Test config flow for Swing2Sleep Smarla integration."""
+
+from unittest.mock import AsyncMock, patch
+
+from homeassistant.components.smarla.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import MOCK_SERIAL_NUMBER, MOCK_USER_INPUT
+
+from tests.common import MockConfigEntry
+
+
+async def test_config_flow(hass: HomeAssistant, mock_connection) -> None:
+    """Test creating a config entry."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
```

Comment:
we also need to mock the async_setup_entry to avoid leaking.

Please search for `mock_setup_entry` and use that as well